warehouse/accounts/views.py

```diff
@@ -206,17 +208,28 @@ def profile(user, request):
     )
 
     # Construct the list of projects with their latest releases from query results
-    projects = [
-        {
+    archived_projects = []
+    live_projects = []
+
+    for row in request.db.execute(stmt):
+        project = {
             "name": row.name,
             "normalized_name": row.normalized_name,
+            "lifecycle_status": row.lifecycle_status,
             "created": row.created,
             "summary": row.summary,
         }
-        for row in request.db.execute(stmt)
-    ]
 
-    return {"user": user, "projects": projects}
+        if row.lifecycle_status == LifecycleStatus.Archived:
+            archived_projects.append(project)
```

Comment:
This conditional is not expressed by tests just yet. Considering this change is also mucking about in the template-land, maybe a functional test is in order.

Likely location: https://github.com/pypi/warehouse/blob/e8195e40986d63f9b72926a67eb6acfd43b36932/tests/functional/test_user_profile.py

I think you've done some client-side HTML assertions in other functional tests before, let me know if you need any pointers!

Reply:
Done! I've added both a functional test and a unit-level one: the functional test confirms that the view doesn't explode when archived projects are present, and the unit test confirms that the `live_projects`/`archived_projects` states are what we expect.

Reply:
Thanks! I appreciate the functional test not-exploding part, but since we have some actual logic encoded in the HTML template like counts and whether or not lists appear, would it make sense to assert those as well? Doesn't need to be exhaustive, but something a bit more useful than "the template doesn't blow up the call".

Reply:
> would it make sense to assert those as well? Doesn't need to be exhaustive, but something a bit more useful than "the template doesn't blow up the call".

I think so, although I don't know how to test those 😅 -- are you thinking we pattern match on the rendered HTML, or something else?

Reply:
Yay, HTML testing time! 
```python
resp = webtest.get(...)

resp.html  # a BeautifulSoup object!!

assert resp.html.h2.text == 


```

Reply:
Oh right, I forgot there was a BS object in the response. Nice, I'll use that.

Reply:
Done -- I've added an assert on the `h2`. I could probably add similar ones on the `package-list` divs, although that goes back to your point about providing more context. So I'm slightly inclined to leave those for a follow-up, unless you feel strongly that they should be included here 🙂

Reply:
This is better than not, and good enough to confirm that the math and pluralization works.